connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -448,7 +459,7 @@ else if (!context.pausedPartitions().isEmpty())
             // Instead of invoking the assignment callback on initialization, we guarantee the consumer is ready upon
             // task start. Since this callback gets invoked during that initial setup before we've started the task, we
             // need to guard against invoking the user's callback method during that period.
-            if (rebalanceException == null) {
+            if (rebalanceException == null || rebalanceException instanceof WakeupException) {
```

Comment:
Note that this fixes a separate bug, which causes us to skip partition open after a rebalance if we get woken up while committing offsets.